src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaHttpUtil.cs

```diff
@@ -0,0 +1,147 @@
+
+            return tags;
+        }
+
+        internal static void SetHttpTagsFromResult(Activity activity, object result)
+        {
+            switch (result)
+            {
+                case APIGatewayProxyResponse response:
+                    activity.SetTag(SemanticConventions.AttributeHttpStatusCode, response.StatusCode.ToString());
+                    break;
+                case APIGatewayHttpApiV2ProxyResponse responseV2:
+                    activity.SetTag(SemanticConventions.AttributeHttpStatusCode, responseV2.StatusCode.ToString());
+                    break;
+            }
+        }
+
+        internal static (string Host, int? Port) GetHostAndPort(string httpScheme, string hostHeaders)
+        {
+            if (hostHeaders == null)
+            {
+                return (null, null);
+            }
+
+            // In case of multiple headres we consider only the 1st.
+            var hostHeader = hostHeaders.Split(',').First();
+            var hostAndPort = hostHeader.Split(new char[] { ':' }, 2);
+            if (hostAndPort.Length > 1)
+            {
+                var host = hostAndPort[0];
+                return int.TryParse(hostAndPort[1], out var port)
+                    ? (host, port)
+                    : (host, null);
+            }
+            else
+            {
+                int? defaultPort = null;
+                switch (httpScheme)
+                {
+                    case "http":
+                        defaultPort = 80;
+                        break;
+                    case "https":
+                        defaultPort = 443;
+                        break;
+                }
+
+                return (hostAndPort[0], defaultPort);
+            }
+        }
+
+        private static string GetHeaderValue(APIGatewayProxyRequest request, string name)
+        {
+            var multiValueHeader = request.MultiValueHeaders?.GetValueByKeyIgnoringCase(name);
+            if (multiValueHeader != null)
+            {
+                // Multiple values for the same header will be separated by a comma.
+                return string.Join(",", multiValueHeader);
+            }
+
+            return request.Headers?.GetValueByKeyIgnoringCase(name);
+        }
+
+        private static string GetHeaderValue(APIGatewayHttpApiV2ProxyRequest request, string name) =>
+            request.Headers?.GetValueByKeyIgnoringCase(name);
+
+        private static string ConstructQueryString(IDictionary<string, string> requestParameters)
+        {
+            if (requestParameters == null || requestParameters.Count == 0)
+            {
+                return string.Empty;
+            }
+
+            var items = new List<string>();
+            foreach (string name in requestParameters.Keys)
+            {
+                items.Add(string.Concat(name, "=", HttpUtility.UrlEncode(requestParameters[name])));
+            }
+
+            return string.Concat("?", string.Join("&", items.ToArray()));
```

Comment:
It seems more efficient to add directly to a StringBuilder instead of constructing an intermediate `List<string>`.

Reply:
Using StringBuilder requires some extra code for considering "&" for the first (or the last) parameter. With this code all work is done by the Join method.